src/main/java/leets/crazyform/domain/workspace/usecase/WorkspaceRetrievalImpl.java

```diff
@@ -0,0 +1,22 @@
+package leets.crazyform.domain.workspace.usecase;
+
+import leets.crazyform.domain.workspace.domain.Workspace;
+import leets.crazyform.domain.workspace.exception.WorkspaceNotFoundException;
+import leets.crazyform.domain.workspace.repository.WorkspaceRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class WorkspaceRetrievalImpl implements WorkspaceRetrieval {
+    private final WorkspaceRepository workspaceRepository;
+
+    @Override
+    public Workspace getWorkspaceById(UUID workspaceId) throws WorkspaceNotFoundException {
+        return workspaceRepository.findById(workspaceId)
+                .orElseThrow(() -> new WorkspaceNotFoundException("Workspace not found with ID: " + workspaceId));
```

Comment:
위 `Exception` 관련 리뷰에 따라서 인자는 삭제해도 될 것 같습니다.
추가로 Exception 반환은 리뷰 내용처럼 축약할 수 있습니다.

```suggestion
                .orElseThrow(WorkspaceNotFoundException::new);
```